src/carma-streets/ParseKafkaLog.py

```diff
@@ -0,0 +1,45 @@
+import json
+import re
+from pathlib import Path
+from enum import Enum
+from KafkaLogMessage import KafkaLogMessage, KafkaLogMessageType
+
+
+def parse_kafka_logs(inputfile: Path, message_type: KafkaLogMessageType)-> list:
+    """Parse Kafka Topic Logs into a list of KafkaLogMessages
+
+    Args:
+        inputfile (Path): Path to an inputfile
+        message_type (KafkaLogMessageType): Type of KafkaLogMessage to parse from input file
```

Comment:
If they are not the correct message type we will encounter JSON parsing errors. This comment also does not claim it verifies message type. It just indicates that this is the message type in the input file. I can reword it to more clearly indicate that

Reply:
> If they are not the correct message type we will encounter JSON parsing errors.

The function's name and signature implies that it will search the specified log file for messages of only type `message_type`, which is not the case. Instead, this function searches the log for _any_ valid JSON string that comes after `CreateTime` then creates a `KafkaLogMessage` instance of type `message_type`.

An uninformed user could do something like this:

```python
result = parse_kafka_logs(spat_log_path, KafakLogMessageType.BSM)
```
It is unclear what is actually happening or why this is problematic. The alternative is clearer:

```python
# Easier to identify a problem
result = extract_kakfa_msgs_as_type(spat_log_path, KafkaMessageType.BSM)

# Easier to understand what is happening
result = extract_kakfa_msgs_as_type(bsm_log_path, KafkaMessageType.BSM)
```

Reply:
I added as type to the function name